source/extensions/path/uri_template_lib/uri_template.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <string>
+
+#include "source/extensions/path/uri_template_lib/proto/rewrite_segments.pb.h"
+#include "source/extensions/path/uri_template_lib/uri_template_internal.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UriTemplate {
+
+// This library provides functionality to rewrite paths based on templates
+// (https://www.rfc-editor.org/rfc/rfc6570.html). Paths are matches against "match patterns" which
+// capture matched tokens. For example: The match pattern "/foo/{bar}" will match "/foo/cat" and
+// will capture "cat" in the variable "bar".
+//
+// A matched path can then be rewritten with a rewrite pattern.
+// For example: rewrite pattern "/pat/hat/{bar}" would use the captured variable "bar" from above
+// to rewrite "/foo/cat" into "/pat/hat/cat"
+
+enum class RewriteStringKind { Variable, Literal };
+
+struct ParsedSegment {
+  ParsedSegment(absl::string_view value, RewriteStringKind kind) : value_(value), kind_(kind) {}
+  absl::string_view value_;
+  RewriteStringKind kind_;
+};
+
+/**
+ * Returns the safe regex that Envoy understands that is equivalent to the given pattern.
+ */
+absl::StatusOr<std::string> convertPathPatternSyntaxToRegex(absl::string_view path_pattern);
+
+/** Parses the specified pattern into a sequence of segments (which are
+/ * either literals or variable names).
```

Comment:
nit: I think `/**` should go on its own line and each line of the comment should start with ` *` (not `/ *`)/